tla/tlc.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+#
+# Python TLC wrapper script for the CCF project
+# Goals:
+# - No dependencies, no venv, no pip install
+# - Set sensible defaults with an eye on performance
+# - Capture useful switches for CI, debugging
+# - Expose specification configuration through CLI
+# - Provide a useful --help, and basic sanity checks
+
+import os
+import argparse
+import pprint
+import pathlib
+
+DEFAULT_JVM_ARGS = [
+    "-XX:+UseParallelGC",  # Use parallel garbage collection, for performance
+]
+
+
+DEFAULT_CLASSPATH_ARGS = ["-cp", "tla2tools.jar:CommunityModules-deps.jar"]
+
+
+def cli():
+    parser = argparse.ArgumentParser(
+        description="TLC model checker wrapper for the CCF project"
+    )
+
+    # Common options for all commands
```

Comment:
would it make sense to add a `other-tlc-options` flag for passing non-supported options directly to tlc?

Reply:
I started that way, but it's much weaker for validation, it's not possible to check the path is valid, or even work out the path to the spec. Plus there is no way to predict the interaction with other options, so I decided against it.

Reply:
In that case, can we add `-difftrace`? I use it quite a lot

Reply:
Adding a generic mechanism to pass `other-tlc-options` and system properties through this script seems like a good idea, especially considering that TLC has numerous additional [options](https://github.com/tlaplus/tlaplus/blob/bdd991e17a13c3d28417ecdfb753fa7cb92ef0a8/tlatools/org.lamport.tlatools/src/tlc2/TLC.java#L231-L304).

Reply:
Missed those comments, somehow, probably because I had the window opened and responded and closed without a refresh. Adding `-difftrace` seems fine.

I am still not keen to add option forwarding, it's an escaping nightmare in my experience, and the benefits compared to running with `-n` and editing the CLI are marginal.

What I propose instead is emitting the env on a single line when using `-n`, so that it will be possible to append any options by doing this:

```
$ `./tlc.py -n sim consensus/SIMccfraft.tla` -difftrace
```

That's less verbose than `--other-tlc-options` and it completely avoids escaping issues.